### PR TITLE
Minecard

### PR DESCRIPTION
**Pull request by harsh-quadB**
## Description
Please include a summary of the changes and the purpose of this PR.

- What does this PR do?
- Why is it needed?

## Screenshots
Include screenshots for **both desktop and mobile** versions.

### Desktop:
[Add screenshot here]

### Mobile:
[Add screenshot here]

## Approach
Briefly explain the approach taken to implement the changes. 

- How did you solve the problem?
- Any particular design patterns or optimizations?

## Issue Number (if any)
Link to the issue this PR addresses, if applicable:

Fixes #[issue_number]

## Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/game_frontend/src/components/ui/Minecard/index.tsx [hunk 1]
```diff
@@ -0,0 +1,144 @@
+import React, { useState } from 'react';
+import './index.css';
+
+// TypeScript interfaces
+interface CardDataType {
```
Comment (BeginArjun) [inline src/game_frontend/src/components/ui/Minecard/index.tsx:5]: All interfaces should be stored inside `types/[Component]/index.d.ts`

File: src/game_frontend/src/components/ui/Minecard/index.tsx [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+          </div>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+
+
+const MineCardsPage: React.FC = () => {
```
Comment (BeginArjun) [inline src/game_frontend/src/components/ui/Minecard/index.tsx:122]: No need for this component. Pls remove it

File: src/game_frontend/src/components/ui/Minecard/index.tsx [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    subtitle: "You control the price",
+    profitPerHour: 0.3,
+    clickPerHour: 45,
+    level: 0,
+    cost: 2,
+  },
+  // ... other card data 
+];
+
+const Card: React.FC<CardProps> = ({
```
Comment (BeginArjun) [inline src/game_frontend/src/components/ui/Minecard/index.tsx:56]: This should be the default export

File: src/game_frontend/src/components/ui/Minecard/index.tsx [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+  image?: string;
+}
+
+interface TabsProps {
+  activeTab: string;
+  onTabChange: (tab: string) => void;
+}
+
+// Card data structure
+const cardData: CardDataType[] = [
```
Comment (BeginArjun) [inline src/game_frontend/src/components/ui/Minecard/index.tsx:25]: Static data shouldn't be in this file. Kindly remove it or comment it

File: src/game_frontend/src/components/ui/Minecard/index.tsx [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+  clickPerHour: number;
+  level: number;
+  cost: number;
+}
+
+interface CardProps extends Omit<CardDataType, 'id'> {
+  image?: string;
+}
+
+interface TabsProps {
```
Comment (BeginArjun) [inline src/game_frontend/src/components/ui/Minecard/index.tsx:19]: Not specific to this component

File: src/game_frontend/src/components/ui/Minecard/index.tsx [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+  image 
+
+}) => {
+  return (
+    <div className="mine-card-wrapper">
+      <div className="mine-card  card-css">
+        <div>
+          <div className="mine-card-image-container">
+            <img 
+              src={image || 'https://cdn.pixabay.com/photo/2023/06/13/15/05/astronaut-8061095_640.png'}
```
Comment (BeginArjun) [inline src/game_frontend/src/components/ui/Minecard/index.tsx:72]: Why is a default image being used? This shouldn't be here.

File: src/game_frontend/src/components/ui/Minecard/index.tsx [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+
+          <div className="mine-card-divider-container">
+            <div className="mine-card-divider"></div>
+          </div>
+          
+          <div className="mine-card-footer">
+            <span className="mine-card-level">Lvl {level}</span>
+            <div className="mine-card-vertical-divider"></div>
+            <span className="mine-card-cost">
+              <span className="mine-card-icon">ic</span>
```
Comment (BeginArjun) [inline src/game_frontend/src/components/ui/Minecard/index.tsx:110]: Where is `ic` written in Figma?

File: src/game_frontend/src/components/ui/Minecard/index.tsx [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+              </span>
+            </div>
+          </div>
+
+          <div className="mine-card-divider-container">
+            <div className="mine-card-divider"></div>
+          </div>
+          
+          <div className="mine-card-footer">
+            <span className="mine-card-level">Lvl {level}</span>
```
Comment (BeginArjun) [inline src/game_frontend/src/components/ui/Minecard/index.tsx:107]: It should be
```jsx
<span className="mine-card-level">Lvl. {level}</span>
```

File: src/game_frontend/src/components/ui/Minecard/index.tsx [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+  },
+  // ... other card data 
+];
+
+const Card: React.FC<CardProps> = ({ 
+  title, 
+  subtitle, 
+  profitPerHour, 
+  clickPerHour, 
+  level,
```
Comment (BeginArjun) [inline src/game_frontend/src/components/ui/Minecard/index.tsx:61]: default level should be 0

**Review by BeginArjun — CHANGES_REQUESTED** [review]
Kindly first pull Changes from the PR #19 after it gets approved and after then start working on these changes.

**Review by ArjunQBTech — APPROVED** [review]
Great Work!

**Pull request merged**